src/analysis.ts

```diff
@@ -103,12 +103,12 @@ export function utonalFundamental(frequencies: number[], maxDivisor = 23) {
 }
 
 // Interval matrix a.k.a the modes of a scale
-export function intervalMatrix(scale: Scale) {
+export function intervalMatrix(scale: Scale, startIndex: number) {
   const result = [];
-  const degrees = [...Array(scale.size + 1).keys()];
+  const degrees = [...Array(scale.size + 1 - startIndex).keys()];
```

Comment:
I know this is my code, but this is no longer an array of `degrees`.
Just do an extra `for` loop below and collect stuff into  a `const row = []` array.